sui/src/bin/validator.rs

```diff
@@ -109,13 +109,27 @@ async fn main() -> Result<(), anyhow::Error> {
         .join(CONSENSUS_DB_NAME)
         .join(encode_bytes_hex(net_cfg.key_pair.public_key_bytes()));
 
+    // Pass in the newtwork parameters of all authorities
+    let net: Vec<(AuthorityName, String, u16)> = network_config
+        .authorities
+        .iter()
+        .map(|authority| {
+            (
+                *authority.key_pair.public_key_bytes(),
+                authority.host.clone(),
+                authority.port,
+            )
+        })
+        .collect();
+
     if let Err(e) = make_server(
         net_cfg,
         &Committee::from(&network_config),
         network_config.buffer_size,
         &consensus_committee,
         &consensus_store_path,
         &consensus_parameters,
+        Some(net),
```

Comment:
Seems like `make_server` always take the net as parameter. Might as well take network_config.authorities as param and construct net in there to avoid code redundancy

Reply:
I am reluctant to push all the logic in, since I am not entirely familiar with all the call points.